.github/workflows/lint.yml

```diff
@@ -0,0 +1,28 @@
+# Lint tests run on PR
+# but should not run after push to main because reporting
+# these after push is meaningless to the building of the package
+name: Lint
+
+on:
+  pull_request:
+    branches: [master]
+
+jobs:
+  lint:
+    runs-on: ${{ matrix.platform }}
+    strategy:
+      matrix:
+        platform: [ubuntu-latest, macos-latest, windows-latest]
+        python-version: ['3.7', '3.8', '3.9', '3.10']
```

Comment:
Do you know any linter that produces different result depending on the Python version? Otherwise I would tend to only use one version here (maybe 3.7 as the oldest supported one)

Reply:
I had this question in another repo before. The only thing are features that do not exist in older python versions. Since we are not doing anything specific we could stick with Linux and python 3.7